include/ada/url.h

```diff
@@ -171,31 +171,38 @@ namespace ada {
      * Return a string representing the scheme. Note that
      * get_scheme_type() should often be used instead.
      */
-    const std::string& get_scheme() const noexcept {
-      return scheme;
+    std::string_view get_scheme() const noexcept {
+      if(is_special()) { return ada::scheme::details::is_special_list[type]; }
+      // We only move the 'scheme' if it is non-special.
+      return non_special_scheme;
     }
 
     /**
-     * Set the scheme for this URL.
+     * Set the scheme for this URL. The provided scheme should be a valid
+     * scheme string, be lower-cased, not contain spaces or tabs. It should
+     * have no spurious trailing or leading content.
      */
     void set_scheme(std::string&& new_scheme) {
-      scheme = new_scheme;
-      type = ada::scheme::get_scheme_type(scheme);
+      type = ada::scheme::get_scheme_type(new_scheme);
+      // We only move the 'scheme' if it is non-special.
+      if(!is_special()) {
+        non_special_scheme = new_scheme;
+      }
     }
 
     /**
      * Take the scheme from another URL.
      */
     void copy_scheme(ada::url&& u) {
-      scheme = u.scheme;
+      non_special_scheme = u.non_special_scheme;
       type = u.type;
     }
 
     /**
      * Take the scheme from another URL.
      */
     void copy_scheme(const ada::url& u) {
```

Comment:
Can we remove the copy_scheme overload (that is only adding a const qualifier?)

Reply:
One moves, one copies. Look again.

Reply:
Added a comment.